frontend/apps/reader/modules/readerview.lua

```diff
@@ -812,40 +806,6 @@ function ReaderView:onReadSettings(config)
     self:resetLayout()
     local page_scroll = config:readSetting("kopt_page_scroll") or self.document.configurable.page_scroll
     self.page_scroll = page_scroll == 1 and true or false
-    self.highlight.saved = config:readSetting("highlight", {})
-    -- Highlight formats in crengine and mupdf are incompatible.
-    -- Backup highlights when the document is opened with incompatible engine.
```

Comment:
No more compatibility issue once everything (pboxes, page/pos0, don't remember what was causing issues) has been moved into bookmarks ? @hius07 ?

Reply:
Sad to see all of this.

Reply:
@hius07 : please be clearer and explicite. Speak ! :)
If you're not fine with all of this, then I might not be - but I want to understand why.
Is it just because you did spend a lot of hours/days making bookmarks and highlights interact with each others, and a feeling of "if it's not broken, don't fix it", and the fear of things possibly breaking ? And all this time you spent being thrown away ? (Which I can understand :) but is it for the better good or not ? :)
Or is it something else ? Are there more technical/architecture/future issues about this that you can think of/feel? If so, please say.

ie, about this specific bit: is it ok to just remove it (and if yes, that's a win if we can get rid of such a kludge) ? Is this becoming a non-issue once we only have bookmarks ?
Or is it that we are just moving the issue to bookmarks, and this can't just be removed, but has to be implemented or workarounded elsewhere ?
Or may be there's still many things to think about and this PR is not as simple as it looks now - and may be you can't give it more thoughts at the moment (which we obviously can understand :/)

Reply:
In all my comments I have talked about the quality of coding only.

Reply:
> In all my comments I have talked about the quality of coding only.

Code-wise, this particular section shows signs of ageing, with numerous patches accumulated over time out of necessity. I would like it more if we could think of a solid architecture first and code it out.

But, even if we do that, I would have done it by rewriting layer by layer (from outer to inner) along with the tests to keep APIs compatible, or at least, easier for other parts to grasp the breaking changes. Discarding the highlights storage and eventually removing the unnecessary codes to bind them to bookmarks is in mind just the first step to do that.

Personally, KOReader is my first attempt in Lua, and by no means it is a simple project. It will take some time to sink in. The beauty of version control lies in the fact that it works like a state machine and has many checks in many places. We can toy with things with minimal destruction. I hope my coding quality will improve over time. :)

Reply:
I welcome your efforts with all my heart. Maybe start with adding new features, small improvements here and there, and after diving deeper into the architecture (which is not simple) solve such complex tasks as this one?

Reply:
> In all my comments I have talked about the quality of coding only.

> solve such complex tasks as this one?

I still don't understand what you think @hius07 .
This PR is mostly code removal, and the added code is mostly existing code moved around, or following the existing code. There's really no invention, no new naming or datastructures, no new way of looping, no clever or unreadable stuff. Mostly just little nits that you and I can point and that are easily updated.

So, the task does not _look_ too complex : it's mostly having an overview of readerview/highlights/bookmarks, which seems to be there.
May be there are things that @uroybd and me have not foreseen, and that you may have thought about - because that's what I fear, I may miss something that you see.
You can't just _say_ it's too "complex" and scare both of us : show some examples of things we have not think about, make this consistent, make the complexity tastable :) so we end up disgusted and think "yes, it's really too complex" :)
But for now, I don't get you.
(For me, it's not too complex - it's just having the will and the time for testing, that @uroybd seems to have. And I'm usually conservative and not the most adventurous :) 

About this particular bit of code, I can think of this case: the current book that is about to have highlight migrated was opened with MuPDF last (so, highlight_rolling backup'ed), and it's now being opened with cre. We're going to work with the "paging" highlights, which may be are not the most worthy highlights to work with.
So, this might be an issue, and something to think about and solve.
Are there more ?!
Do we want to have to think about them ?
Do we all have the time at this moment to do so ?
And if not, we should at least point them and write about what we see/think of.

Reply:
> About this particular bit of code

I think we should keep it: there is the same kind of backup/restore in readerbookmark.lua.
One can open an EPUB with crengine or with MuPDF (long-press, `Open with...`). Bookmarks also are incompatible, and could cause crashes. In the past, we were deleting the other-type ones, but @hius07 added this so we are gentle and they are backup/restored instead of being deleted.
So, this will continue to happen for Bookmarks.

But when migrating, we would want to migrate both, and the 2 migrations could happen at different  moments. So, either:
- keep this, and do the migration highlight>bookmarks after - keeping in mind that on a future re-opening of this book with the other engine, the `highlight_rolling/paging` will be restored and renamed to `highlight` (and so, `bookmarks_rolling/paging` too, and all this should all happen before we do the highlight migration).
- do both migration (if it happens the docsettings have the 2 kinds of bookmarks/highlights) at once (but this feels complicated, and the need to juggle with readerbookmarks so it also reads the old one if we want to use its methods...)